components/camel-file/src/main/java/org/apache/camel/component/file/GenericFileConsumer.java

```diff
@@ -621,6 +621,26 @@ private boolean notUnique(GenericFile<T> file) {
         return false;
     }
 
+    /**
+     * Strategy to perform hidden file matching based on endpoint configuration.
+     * <p/>
+     * Will always return <tt>false</tt> for certain files/folders:
+     * <ul>
+     * <li>Starting with a dot (hidden)</li>
+     * </ul>
+     */
+    protected boolean isMatchedHiddenFile(GenericFile<T> file, boolean isDirectory) {
+        String name = file.getFileNameOnly();
+
+        // folders/names starting with dot is always skipped (eg. ".", ".camel",
```

Comment:
This is not desirable. I think we can skip whatever is created by camel, but we should be able to upload contents of other hidden folders.